src/engine/sched.c

```diff
@@ -1496,9 +1496,8 @@ sched_req_sleep(struct sched_request *req, uint32_t msecs)
 static void
 req_wakeup_internal(struct dss_xstream *dx, struct sched_request *req)
 {
-	D_ASSERT(req != NULL);
 	/* The request is not in sleep */
-	if (req->sr_wakeup_time == 0)
+	if (req == NULL || req->sr_wakeup_time == 0)
```

Comment:
Because under check mode, some background ULT may be not started, then when we stop the pool/container, related XXX_req may be NULL. It is possible to make the caller to check whether XXX_req is NULL or not before calling sched_req_wakeup(), but checking that inside req_wakeup_internal() is more convenient.